### ISSUE TITLE
[mydumper] - regression - 0.13 - freeze at the step "Schema Done, Starting Non-Innodb" (--no-data)

### ISSUE THREAD

**Issue by anthosz** [mydumper/mydumper#926]
**Describe the bug**
Once the schema is dumped (without datas), it start to do something (Starting Non-Innodb) but I don't know what... In the database, we can see 9 connections in the processlist that "sleep" until it's timeout

**To Reproduce**
Command executed:
* mydumper --no-data --threads 8 --defaults-file=${CONFILE} --compress --verbose 3 --outputdir ${PATH}

What mydumper and myloader version has been used?
mydumper0.13.1-1, built against MySQL 5.7.40-43 with SSL support with GZIP

**Expected behavior**
It stop once the schema dump is done

**Log**
```
** Message: 11:42:22.814: Thread 7 dumping schema for `db`.`a`
** Message: 11:42:22.814: Thread 8 dumping schema for `db`.`b`
** Message: 11:42:22.815: Thread 1 dumping schema for `db`.`c`
** Message: 11:42:22.815: Thread 4 dumping schema for `db`.`d`
** Message: 11:42:22.815: Thread 5 dumping schema for `db`.`e`
** Message: 11:42:22.815: Thread 6 dumping schema for `db`.`f`
** Message: 11:42:22.815: Thread 2 dumping schema for `db`.`g`
** Message: 11:42:22.815: Thread 3 dumping schema for `db`.`h`
** Message: 11:42:22.816: Thread 1 Schema Done, Starting Non-Innodb
** Message: 11:42:22.815: Thread 8 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 4 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 2 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 5 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 3 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 7 Schema Done, Starting Non-Innodb
** Message: 11:42:22.816: Thread 6 Schema Done, Starting Non-Innodb
```

**Environment (please complete the following information):**
 - Ubuntu 20.04OS version: [e.g. CentOS 7.9, Ubuntu 20.04]
 - RDS Servless V2 (AWS / Mysql V8)

**Additional context**
It worked well with the 0.12 version

**Comment by kdckrs** [comment]
Can confirm I am having the same issue.

**Comment by acjandupont** [comment]
I can confirm this issue, having the same problem.

**Comment by davidducos** [comment]
I already added the test on the branch and it is failing.

**Comment by pierrexp9** [comment]
I confirm this is still happening on 0.13.1-1
I had to go back to 0.12.7-3 to have mydumper work again.

This should not be closed

**Comment by pierrexp9** [comment]
I confirm this is still happening on 0.13.1-1
I had to go back to 0.12.7-3 to have mydumper work again.

I understand this will be fixed into 0.13.1-2, but should this issue be closed until 0.13-1-2 is out ?
